### ISSUE TITLE
Get different results in eager and graph mode when I use tf.keras.Reduction.NONE on Loss object.

### ISSUE THREAD

**Issue by fsx950223** [tensorflow/tensorflow#37656]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): 
- OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04): Ubuntu
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if
the issue happens on mobile device: 
- TensorFlow installed from (source or
binary): - TensorFlow version (use command below): 
- Python version: - Bazel
version (if compiling from source):2.2.0
- GCC/Compiler version (if compiling from
source): 
- CUDA/cuDNN version: - GPU model and memory:

[Colab example](https://colab.research.google.com/drive/1VWVty280phZLVsmB91aW6GqdQoEgDCdP)

Related [addons](https://github.com/tensorflow/addons/issues/1320)

**Comment by ravikyram** [comment]
I have tried on colab with TF version 2.2.0-rc0 , 2.2.0-dev20200316 and was able to reproduce the issue.Please, find the gist [here.](https://colab.sandbox.google.com/gist/ravikyram/b40935f0f80653729e4ef90294b47f1c/untitled732.ipynb) Thanks!

**Comment by jvishnuvardhan** [comment]
@fsx950223 The tensorflow doc on TF website describes the reported loss value will be a scalar, but unreduced vector will be passed to the optimizer.

> NONE: Weighted losses with one dimension reduced (axis=-1, or axis specified by loss function). When this reduction type used with built-in Keras training loops like fit/evaluate, the unreduced vector loss is passed to the optimizer but ***the reported loss will be a scalar value***.

Based on this description on [TF website](https://www.tensorflow.org/api_docs/python/tf/keras/losses/Reduction), you are getting a scalar value (`30.75`) which is correct right? Thanks!

**Comment by fsx950223** [comment]
> @fsx950223 The tensorflow doc on TF website describes the reported loss value will be a scalar, but unreduced vector will be passed to the optimizer.
> 
> > NONE: Weighted losses with one dimension reduced (axis=-1, or axis specified by loss function). When this reduction type used with built-in Keras training loops like fit/evaluate, the unreduced vector loss is passed to the optimizer but _**the reported loss will be a scalar value**_.
> 
> Based on this description on [TF website](https://www.tensorflow.org/api_docs/python/tf/keras/losses/Reduction), you are getting a scalar value (`30.75`) which is correct right? Thanks!

Yes, it's correct. But graph and eager mode should return same result when I use ```evaluate```.

**Comment by ravikyram** [comment]
I have tried in TF nightly version(`2.5.0-dev20201029`) and was able to reproduce the issue. Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/bf520e40b8fedb7e93840d2fc1e5501f/untitled96.ipynb?authuser=1).Thanks!

**Comment by sachinprasadhs** [comment]
Was able to reproduce your issue in Tf Nightly 2.6.0-dev20210524, please find the gist [here](https://colab.research.google.com/gist/sachinprasadhs/055ff3a8ee0b28db2352a0ddc5d96721/untitled7.ipynb). Thanks!

**Comment by kumariko** [comment]
Was able to reproduce issue in Tf 2.7, please find the gist [here](https://colab.research.google.com/gist/kumariko/99780d7b0e651df8f45be967ddb68b82/untitled7.ipynb#scrollTo=R-FZoQlKLuxr). Thanks!

**Comment by tensorflowbutler** [comment]
Hi There,

 This is a stale issue. As you are using an older version of tensorflow, we are checking to see if you still need help on this issue. Please test the issue with the latest TensorFlow (TF2.7 and tf-nightly). If the issue still persists with the newer versions of TF, please feel free to open it in [keras-team/keras](https://github.com/keras-team/keras/issues) repository by providing details about the issue and a standalone code to reproduce the issue. Thanks! 

 Please note that Keras development has moved to a separate Keras-team/keras repository to focus entirely on only Keras. Thanks!
